ga/src/main/scala/cilib/ga/RandomSearch.scala

```diff
@@ -0,0 +1,28 @@
+package cilib
+package ga
+
+import cilib.Lenses._
+import scalaz.std.list._
+import scalaz.syntax.traverse._
+
+// Author: Kyle Erwin
+object RandomSearch {
+
+    def ga[S](parentSelection: List[Individual] => RVar[List[Individual]], deviation: Double): List[Individual] => Individual => Step[Double, List[Individual]] =
+        collection => x => for {
+            parents <- Step.pointR(parentSelection(collection))
+            mutated <- Step.pointR[Double, List[Individual]](mutation(parents, deviation))
+            evaluated <- mutated.traverseU(x => Entity.eval((v: Position[Double]) => v)(x))
+        } yield evaluated
+
+    // The mutation method is applied to all the parents.
+    // The parents are mutated by adding a random value gaussian distribution to
+    // point in their position.
+    def mutation(parents: List[Individual], deviation: Double): RVar[List[Individual]] = {
```

Comment:
So, mutation can be generalised a little more, right?

I'd prefer us to have a function in the form `A => RVar[A]`, whereby you pass in a Distribution. This way you can pass in the whatever kind of distribution to apply, not just gaussian.

The function that then be applied onto the collection, via a `traverse`

Reply:
Sure. I'll get working on that.